crates/uv-resolver/src/lock.rs

```diff
@@ -1033,3 +1120,43 @@ impl std::fmt::Display for HashParseError {
         self.0.fmt(f)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn hash_required_present() {
+        let data = r#"
+version = 1
+
+[[distribution]]
+name = "anyio"
+version = "4.3.0"
+source = "registry+https://pypi.org/simple"
+
+[[distribution.wheel]]
+url = "https://files.pythonhosted.org/packages/14/fd/2f20c40b45e4fb4324834aea24bd4afdf1143390242c0b33774da0e2e34f/anyio-4.3.0-py3-none-any.whl"
+"#;
+        let result: Result<Lock, _> = toml::from_str(data);
+        insta::assert_debug_snapshot!(result);
+    }
+
+    #[test]
+    fn hash_required_missing() {
+        let data = r#"
+version = 1
+
+[[distribution]]
+name = "anyio"
+version = "4.3.0"
+source = "path+file:///foo/bar"
+
+[[distribution.wheel]]
+url = "file:///foo/bar/anyio-4.3.0-py3-none-any.whl"
+hash = "sha256:048e05d0f6caeed70d731f3db756d35dcc1f35747c8c403364a8332c630441b8"
```

Comment:
I wrote this lock file just to provoke an error where a hash was found but wasn't expected.

But I'm not sure this is right. And has led me to wonder what a path dependency with a wheel should actually look like in the lock file. Notice, for example, the repetition in the `source` field above with the `url` in the `wheel` entry. The data model currently allows, for example, a path dependency to have multiple wheels.

(This issue isn't really related to this PR, but the lock file here is odd and I wanted to call it out.)